File: tests/fixtures/sample_repo_python/src/sample_app/service.py
from collections.abc import Callable

from .models import User
from .repository import UserRepository


def audited(func: Callable[[object, str], User]) -> Callable[[object, str], User]:
	return func


class UserService:
	def __init__(self) -> None:
		self.repository = UserRepository()

	@staticmethod
	def normalize(name: str) -> str:
		return name.strip()

	@audited
	def load(self, name: str) -> User:
		try:
			user = self.repository.get(self.normalize(name))
			return user
		except ValueError as exc:
			raise RuntimeError('could not load user') from exc

File: tests/fixtures/sample_repo_typescript/src/service.ts
import { logged } from './decorators';
import { UserRepository } from './repository';

export class UserService {
  constructor(private readonly repository = new UserRepository()) {}

  @logged
  async load(id: string) {
    try {
      const user = await this.repository.get(id);
      return user;
    } catch (error) {
      throw new Error('could not load user', { cause: error });
    }
  }
}
